samples/nrf9160/modem_shell/overlay-lwm2m_client_pgps.conf

```diff
@@ -0,0 +1,37 @@
+#
+# Copyright (c) 2022 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+# LwM2M P-GPS overlay configuration, to be used in combination with the LwM2M client overlay
+
+# LwM2M client
+CONFIG_LWM2M_CLIENT_UTILS_LOCATION_ASSIST_PGPS=y
+
+# nRF Cloud P-GPS
+CONFIG_NRF_CLOUD_PGPS=y
+CONFIG_NRF_CLOUD_PGPS_REPLACEMENT_THRESHOLD=4
+CONFIG_NRF_CLOUD_PGPS_TRANSPORT_NONE=y
+CONFIG_NRF_CLOUD_PGPS_DOWNLOAD_TRANSPORT_CUSTOM=y
+CONFIG_NRF_CLOUD_PGPS_REQUEST_UPON_INIT=n
+
+# Storage for P-GPS
+CONFIG_STREAM_FLASH=y
+CONFIG_FLASH=y
+CONFIG_FLASH_PAGE_LAYOUT=y
+CONFIG_FLASH_MAP=y
+CONFIG_FCB=y
+CONFIG_SETTINGS=y
+CONFIG_SETTINGS_FCB=y
+CONFIG_MPU_ALLOW_FLASH_WRITE=y
```

Comment:
We have already overlay for p-gps (overlay-pgps.conf), and most of the above are there. 
IMO, that file should be used as a base for common p-gps stuff and this file should add the p-gps stuff specific with lwm2m_client.
Usage would be always like:
-DOVERLAY_CONFIG="overlay-pgps.conf;overlay-wifi-location-nrf-cloud-mqtt.conf"

Reply:
True, most of the stuff is in overlay-pgps.conf. I'll remove the generic stuff and update the instructions.

@trantanen suggested earlier that we could remove the word "client" from the LwM2M overlays. I'm also now leaning towards this, so I'll change the overlay file names. The word "client" isn't really necessary.